### ISSUE TITLE
The solution for accuracy reward

### ISSUE THREAD

**Issue by markinruc** [tulerfeng/Video-R1#36]
I noticed the dataset preparation function make_conversation_image_and_video(example) in grpo.py

`
msg ={
            "prompt": 
               [{
                    "role": "user",
                    "content": [
                        {
                            "type": example['data_type'],
                            # example['data_type']: os.getcwd() + "/Video-R1-data" + example['path'][1:]
                        },
                        {
                            "type": "text",
                            "text": QUESTION_TEMPLATE.format(Question=question) + TYPE_TEMPLATE[example['problem_type']]
                        }
                        ]
                }]
            }
`

It seems that the answer of each sample is not passed into the msg. How can we get the solution in the accuracy reward function?

Maybe I have not found the right usage of the grpo trainer so I am quite confused.

Thanks  a lot!

**Comment by tulerfeng** [comment]
Thanks for your quension.

Although answers are not passed into msg, they are stored in the dataset.

In `grpo_trainer.py`, we use the following code to call the reward function:
```
        # Compute the rewards
        prompts = [prompt for prompt in prompts for _ in range(self.num_generations)]
        rewards_per_func = torch.zeros(len(prompts), len(self.reward_funcs), device=device)
        for i, (reward_func, reward_processing_class) in enumerate(
            zip(self.reward_funcs, self.reward_processing_classes)
        ):
            # Repeat all input columns (but "prompt" and "completion") to match the number of generations
            reward_kwargs = {key: [] for key in inputs[0].keys() if key not in ["prompt", "completion"]}
            for key in reward_kwargs:
                for example in inputs:
                    # Repeat each value in the column for `num_generations` times
                    reward_kwargs[key].extend([example[key]] * self.num_generations)
            output_reward_func = reward_func(prompts=prompts, completions=completions, **reward_kwargs)
            rewards_per_func[:, i] = torch.tensor(output_reward_func, dtype=torch.float32, device=device)
```

The groud_truth here in in the `solution` key of `reward_kwargs`


While the reward function (used for computing the reward) is in `grpo.py`:

```
def accuracy_reward(completions, solution, **kwargs):
        
......
    question_type = kwargs['problem_type'][0]
    
    contents = [completion[0]["content"] for completion in completions]
    current_time = datetime.now().strftime("%d-%H-%M-%S-%f")
    rewards = []

    for content, sol in zip(contents, solution):
    
        try:
            output_ans = extract_answer(content)
            gt_ans = extract_answer(sol)
......

            
    return rewards
```
 
The ground truth answer is in `gt_ans = extract_answer(sol)`

**Comment by markinruc** [comment]
Thanks a lot!
